cmake/FindCharm.cmake

```diff
@@ -19,34 +19,64 @@ if (NOT EXISTS "${CHARM_ROOT}")
 endif ()
 
 find_path(
-    CHARM_INCLUDE_DIRS charm.h
+    CHARM_INCLUDE_DIR charm.h
     PATH_SUFFIXES include
     HINTS ${CHARM_ROOT} ENV CHARM_ROOT
 )
+set(CHARM_INCLUDE_DIRS ${CHARM_INCLUDE_DIR})
```

Comment:
Why change the name of the variable, then assign it to the original name on the next line?

Reply:
The `CHARM_INCLUDE_DIR` is what's actually found by cmake (a single directory). The `CHARM_INCLUDE_DIRS` is only a variable that's defined by convention and used in `SetupCharm.cmake`. It can be deleted once we have switched charm to an imported target. Makes no sense to write `${CHARM_INCLUDE_DIRS}/charm-version.h`.

Reply:
But isn't `CHARM_INCLUDE_DIRS` deterministically set to the value of `CHARM_INCLUDE_DIR` in this line? Is it changed in `SetupCharm.cmake`?
I guess what I'm saying is that it seems confusing to have two variables floating around that differ by one character and encode the same information -- if the version without the `S` is preferable, why not just change it everywhere?

Reply:
None of this is preferable, the FindCharm shouldn't define variables "by convention" at all :D It should define imported targets. I do that in #2680.

Reply:
I just didn't want to write `${CHARM_INCLUDE_DIRS}/charm-version.h`, so I defined `${CHARM_INCLUDE_DIR}` instead but kept the old name so I didn't have to change the other file. I think old cmake code exposing these conventional variables generally work with the `S` to have a list of dirs that is added to some global list of include dirs.